mage.go

```diff
@@ -0,0 +1,49 @@
+//go:build mage
+
+package main
+
+import (
+	"fmt"
+	"github.com/magefile/mage/sh"
```

Comment:
Why not use https://pkg.go.dev/os/exec#Cmd.Run which is a go native library instead of a 3rd party one?

Reply:
This contains a wrapper that builds the environment conveniently with just a hashmap of environment variables; we depend on `mage` to run the file anyhow, so we might as well leverage the niceties they include; the actual code that is executed when setting up the environment can be seen [here](https://github.com/magefile/mage/blob/526bf4775130d567de6b18fea8f3791d76c8be2a/sh/cmd.go#L126).